cppcache/integration-test/QueryHelper.hpp

```diff
@@ -315,76 +315,58 @@ void QueryHelper::getPDXObject(std::shared_ptr<Region>& rptr) {
                            size_t expectedRows) {
-  if (!std::dynamic_pointer_cast<ResultSet>(resultSet)) {
-    return false;
-  }
-
-  auto rsptr = std::static_pointer_cast<ResultSet>(resultSet);
-
-  size_t foundRows = 0;
-
-  SelectResultsIterator iter = rsptr->getIterator();
+  if (auto rsptr = std::static_pointer_cast<ResultSet>(resultSet)) {
+    size_t foundRows = 0;
+    for (auto&& row : *rsptr) {
+      foundRows++;
+    }
 
-  for (size_t rows = 0; rows < rsptr->size(); rows++) {
-    auto ser = (*rsptr)[rows];
-    foundRows++;
+    printf("found rows %zd, expected %zd \n", foundRows, expectedRows);
+    if (foundRows == expectedRows) return true;
   }
-
-  printf("found rows %zd, expected %zd \n", foundRows, expectedRows);
-  if (foundRows == expectedRows) return true;
-
   return false;
 }
 
 bool QueryHelper::verifySS(std::shared_ptr<SelectResults>& structSet,
                            size_t expectedRows, int32_t expectedFields) {
-  if (!std::dynamic_pointer_cast<StructSet>(structSet)) {
-    if (expectedRows == 0 && expectedFields == 0) {
-      return true;  // quite possible we got a null set back.
-    }
-    printf("we have structSet itself nullptr \n");
-    return false;
-  }
-
-  auto ssptr = std::static_pointer_cast<StructSet>(structSet);
+  if (auto ssptr = std::static_pointer_cast<StructSet>(structSet)) {
+    size_t foundRows = 0;
+    for (auto&& ser : *ssptr) {
+      foundRows++;
 
-  size_t foundRows = 0;
+      auto siptr = std::dynamic_pointer_cast<Struct>(ser);
 
-  for (SelectResults::Iterator iter = ssptr->begin(); iter != ssptr->end();
-       iter++) {
-    auto ser = (*iter);
-    foundRows++;
+      if (siptr == nullptr) {
+        printf("siptr is nullptr \n\n");
+        return false;
+      }
 
-    Struct* siptr = dynamic_cast<Struct*>(ser.get());
+      int32_t foundFields = 0;
+      for (auto&& field : *siptr) {
+        foundFields++;
+      }
 
-    if (siptr == nullptr) {
-      printf("siptr is nullptr \n\n");
-      return false;
+      if (foundFields != expectedFields) {
+        char buffer[1024] = {'\0'};
+        sprintf(buffer, "found fields %d, expected fields %d \n", foundFields,
+                expectedFields);
+        LOG(buffer);
+        return false;
+      }
     }
 
-    int32_t foundFields = 0;
+    if (foundRows == expectedRows) return true;
 
-    for (int32_t cols = 0; cols < siptr->length(); cols++) {
-      auto field = (*siptr)[cols];
-      foundFields++;
-    }
-
-    if (foundFields != expectedFields) {
-      char buffer[1024] = {'\0'};
-      sprintf(buffer, "found fields %d, expected fields %d \n", foundFields,
-              expectedFields);
-      LOG(buffer);
-      return false;
+    // lets log and return in case of error only situation
+    char buffer[1024] = {'\0'};
+    sprintf(buffer, "found rows %zd, expected rows %zd\n", foundRows,
+            expectedRows);
+    LOG(buffer);
+  } else {
+    if (expectedRows == 0 && expectedFields == 0) {
```

Comment:
Won't we have already returned from the `if (foundRows == expectedRows)` check above?

Reply:
Ack, that's what I get for eyeballing it :).